zebra/label_manager.c

```diff
@@ -96,40 +142,14 @@ int zread_relay_label_manager_request(int cmd, struct zserv *zserv)
 	if (cmd == ZEBRA_RELEASE_LABEL_CHUNK)
 		return 0;
 
-	/* read response */
-	src = zclient->ibuf;
-	dst = zserv->obuf;
-
-	stream_reset(src);
-
-	u_int16_t size;
-	u_char marker;
-	u_char version;
-	vrf_id_t vrf_id;
-	u_int16_t resp_cmd;
-	ret = zclient_read_header(src, zclient->sock, &size, &marker, &version,
-				  &vrf_id, &resp_cmd);
-	if (ret < 0) {
-		zlog_err("%s: Error reading label chunk response: %s", __func__,
-			 strerror(errno));
+	/* wait for response */
+	FD_ZERO (&readfds);
+	FD_SET (zclient->sock, &readfds);
+	select (zclient->sock+1, &readfds, NULL, NULL, NULL);
```

Comment:
Although this is only the relay case, where zebra acts as a proxy to an external label manager. I guess we could do this one async, and keep synchronous only clients. You are totally right that, while it makes sense on client side, this is dangerous for zebra.
I'll try it and push a commit here.